### PR TITLE
Adding pkgdown site built by GHA

### PR DESCRIPTION
**Pull request by apreshill**
Leaving as a draft for now while we finalize the pkgdown template design (cc @cderv)

This site currently publishes to: https://pkgs.rstudio.com/rmarkdown/

The site is in the RStudio Netlify org.

To do before merging:

1. Finalize pkgdown template (adding book to homepage)

1. Change the branch to deploy from in the GH action to `master`

1. Add 2 secrets (`NETLIFY_AUTH_TOKEN` and `NETLIFY_SITE_ID`) to this repo. I don't have admin access to add myself.

1. Add a code of conduct (required by [Netlify's open source policy](https://www.netlify.com/legal/open-source-policy/))

Note: we currently depend on the [dev version](https://github.com/r-lib/pkgdown/commit/6094ac3dd80a1f3f1d4e2f47d0ed4716d51362b0) of pkgdown to render `NEWS.md` correctly.

### CODE REVIEW

File: .github/workflows/pkgdown.yaml
Comment (cderv) [inline .github/workflows/pkgdown.yaml:57]: Just to share Netlify as an action for this : https://github.com/netlify/actions/tree/master/cli

File: .github/workflows/pkgdown.yaml
Comment (cderv) [inline .github/workflows/pkgdown.yaml:47]: I don't think we need this. There is no commit to make as we deploy on netlify and not github pages

**Comment by cderv** [comment]
What is missing to move ahead for a first version online on the new website ? 

I think currently we are on a weird state with: 

* https://pkgs.rstudio.com/rmarkdown/ online but very old
* https://rmarkdown.rstudio.com/docs/ that was no more updated (but fixing that)

So we got two sites and I don't think that is ideal - or at least they should be in sync regarding the version.
I know we've been busy on other topic but we should move out from this in-between state.  

One generic remark about our pkgdown site : How do we position regarding dev version ? 

* CRAN version will always be X.Y
* Dev version will be X.Y.Z

Are we okay with only one site as we have currently ? Or do we need to get two sites ? 
**pkgdown** has a development mode to allow a release site in `docs/` and devel site in `docs/devel`. There is mode auto based on the version in description but it works only with a convention of `x.y.z` for a released version and `x.y.z.*` for a development version - which won't work for us. So that would require a non automatic workflow.

File: .github/workflows/pkgdown.yaml
Comment (cderv) [reply .github/workflows/pkgdown.yaml:57]: But it is working only on linux... so this remark above do not matter

File: .github/workflows/pkgdown.yaml
Comment (cderv) [inline .github/workflows/pkgdown.yaml:57]: I deactivated this part because I do not want the PR to overwrite the current website. 

@apreshill I need your skill with netlify (and because I don't think I have access to RStudio Netlify org) so that we have a preview from this PR (and not just from your fork)

Can we have a preview deploy ?

**Comment by cderv** [comment]
I think this is now ready. If I take the checklist from your first post

- [x] Finalize pkgdown template (adding book to homepage)
- [ ] Change the branch to deploy from in the GH action to master => this will be done before merging
- [x] Add 2 secrets (NETLIFY_AUTH_TOKEN and NETLIFY_SITE_ID) to this repo. I don't have admin access to add myself. => Secrets are currently there, but used to deploy to https://rmarkdown.rstudio.com/docs/ . So this will require a switch or a parallel run of both site
- [ ] Add a code of conduct (required by Netlify's open source policy) - about this, we could use `usethis::use_code_of_conduct(".github")`. It would add a `.github/CODE_OF_CONDUCT.md` that would be the one from https://www.contributor-covenant.org/version/2/0/code_of_conduct/. See examples in https://usethis.r-lib.org/CODE_OF_CONDUCT.html - it would be link on the sidebar in homepage e.g https://usethis.r-lib.org/index.html. I am ok with adding this to our website + a word about it at the bottom of the README. 

During the work session with @apreshill we discussed a bit https://github.com/rstudio/rmarkdown/pull/1955#issuecomment-745201211. Currently, the versionning of our packages is not supported by the [mode auto for development](https://pkgdown.r-lib.org/reference/build_site.html#development-mode). This means that currently our site is build for the development version always. So, about that: 
* I added the correct tooltip for this (b35b826d99473fd1f8872388be4aa68e253e3bdf). I did not change the color because **quillt** does not currently have correct color for `info`, `warning` `danger`. We could change that in the template @apreshill ? 
* I will try to change the workflow for having https://pkgs.rstudio.com/rmarkdown/ and https://pkgs.rstudio.com/rmarkdown/dev/ but do that later after this PR and others are merged 

A part from the both uncheck box above, I think this is ready to be merged now. We just need you final review @yihui and come up with a switch plan with the two websites. 

As a reminder the published version of this branch is here: https://pkgs.rstudio.com/rmarkdown/

**Review by yihui — APPROVED** [review]
> * Add 2 secrets (NETLIFY_AUTH_TOKEN and NETLIFY_SITE_ID) to this repo. I don't have admin access to add myself. => Secrets are currently there, but used to deploy to https://rmarkdown.rstudio.com/docs/ . => **this will require a switch of SITE ID or a parallel run of both sites with two different secrets**

I have added these secrets and the site should be published to `rmarkdown-pkg.netlify.app` now. With d52322f, https://rmarkdown.rstudio.com/docs/ should also point to this site. Previously, it pointed to `rmarkdown-docs.netlify.com`, and this site will be retired.

> Currently, the versionning of our packages is not supported by the [mode auto for development](https://pkgdown.r-lib.org/reference/build_site.html#development-mode).

I feel **pkgdown**'s auto mode could have been able to detect if a version is a dev version regardless of the version number scheme (i.e. not requiring authors to use `x.y.z.9000`). For example, when building NEWS.md, it knows that a version is unreleased: https://pkgs.rstudio.com/rmarkdown/news/index.html#rmarkdown-2-8-unreleased

Anyway, that's not super important to me, and the tooltip is good enough. I don't really care. I prefer not having both  `https://pkgs.rstudio.com/rmarkdown/` and `https://pkgs.rstudio.com/rmarkdown/dev/`.

> * Add a code of conduct (required by Netlify's open source policy)

That is required only for users of their open source plan. We use their Pro plan, so this requirement doesn't really apply to us. That said, I'm okay with adding a link to https://www.contributor-covenant.org/version/2/0/code_of_conduct/.

Thanks!

**Comment by apreshill** [comment]
Just a note:
        Add a code of conduct (required by Netlify's open source policy)

That is required only for users of their open source plan. We use their Pro plan, so this requirement doesn't really apply to us. That said, I'm okay with adding a link to https://www.contributor-covenant.org/version/2/0/code_of_conduct/.

---

Yes, this was because we were planning to move our pkgdown sites for open source packages to an open source org, off the Pro account.

**Comment by yihui** [comment]
Oh I forgot to mention one minor thing yesterday: make sure the PNGs are optimized (you can `brew install optipng` and run `xfun::optipng()`). Same thing for https://github.com/rstudio/blogdown/pull/511 and https://github.com/rstudio/bookdown/pull/1028.

**Comment by cderv** [comment]
Yes I'll recheck that but I did use `webshot2::shrink()` which does the same.

**Comment by yihui** [comment]
Okay. We also have other PNG files (e.g., the hex logos and icons).

File: .github/workflows/pkgdown.yaml
Comment (cderv) [inline .github/workflows/pkgdown.yaml:61]: To remove before merge

**Comment by cderv** [comment]
@apreshill I believe this is now ready. Yihui has set the secrets to prepare the switch so we are good. 
There is just the contact email for CoC to add once it is confirmed. 

Is it good for you for this one ? Is this ok to merge ? Thanks.

**Comment by apreshill** [comment]
I think so!

**Pull request merged**
